### ISSUE TITLE
Analyzer Question

### ISSUE THREAD

**Issue by fujiiface** [MarimerLLC/cslaforum#403]
I recently started upgrading to CSLA 4.6.603 and noticed some analyzer errors that I was not previously getting.  One of them is `Managed backing fields must be public, static, and read-only`.

Does this only apply to editable objects?  I have a `ReadOnlyBase `class that does not have a `public static readonly` managed property and I do not get the error for those objects but I do get it for my `BusinessBase `objects.

**Comment by ajj7060** [comment]
This should apply to all Csla stereo types.  

I think the reason is that on some platforms reflection is limited to reflecting over public members of the type only, and Csla needs to use reflection to find the PropertyInfos.  

They need to be static and readonly b/c the object reference is used internally by Csla as dictionary keys, for one.

**Comment by rockfordlhotka** [comment]
This might be a bug in the analyzer - @JasonBock would know.

**Comment by fujiiface** [comment]
@ajj7060 @rockfordlhotka @JasonBock So, even though the analyzer doesn't throw the error it's safe to say that I should go through my ReadOnlyBase classes and update them?

**Comment by rockfordlhotka** [comment]
Yes, in all cases the static property field should look like this (from the standard snippet):

```c#
    public static readonly PropertyInfo<string> NameProperty = 
      RegisterProperty<string>(c => c.Name);
    public string Name
    {
      get { return GetProperty(NameProperty); }
      set { SetProperty(NameProperty, value); }
    }
```

**Comment by fujiiface** [comment]
Thanks @rockfordlhotka!

**Comment by JasonBock** [comment]
Wait, why was this closed? I never had a chance to look at the analyzer and see if there's a bug there (been way too busy with being on vacation the past 5 days :)). Let's re-open it please until I determine that there truly is an issue.

**Comment by fujiiface** [comment]
Reopened.  Sorry didn't think it was an issue but I'll leave it open until you say otherwise.  Thanks for looking into it.
